subprojects/messaging/src/main/java/org/gradle/internal/serialize/kryo/StringDeduplicatingKryoBackedEncoder.java

```diff
@@ -114,4 +105,95 @@ public void close() {
     public void done() {
         strings = null;
     }
+
+    private class IndexedStringSet {
```

Comment:
I think some explanatory Javadoc would be helpful here, and in the inner classes below. It took me a little while to work out what was going on :).

Reply:
Docs added.